src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -101,6 +103,65 @@ Mass Segment::Solution::computeDeltaMass() const
     return Mass::Kilograms(getInitialMass().inKilograms() - getFinalMass().inKilograms());
 }
 
+Array<MatrixXd> Segment::Solution::computeAccelerations() const
```

Comment:
Nice! just few comments:

I would update the signatrue to:

```
MatrixXd Segment::Solution::getContributionsconst Shared<Dynamics> aDynamcis, const Shared<Frame> aFrame)
```

- `aDynamics` allows us to only re-compute the dynamic we are interested in.
- `aFrame` allows us to specify the output frame in which we would like the contribution to be expressed in. This is not a big rework the only thing you'd need to add is a `state.inFrame(aFrame)` conversion before you to the rest.

On the naming:
- I would go from `Acceleration` to `Contribution` to keep it generic and flexible.
- I know that at the moment this is re-computing, but I would still call this `get` rather than `compute`. The reason being that hopefully, at some point, we don't need to recompute* this here (as it happens during segment solving), then we can start using `get` already and not break the API later on.

*Additional comment on re-computing (not needed or anything, just a suggestion):

If I were to call this twice for the same dynamic, and the same frame, we will be computing it twice, if I then call it again, then three times. Maybe you can have an internal `Map<Dynamics,MatrixXd>` that starts empty. And whenever you're asked the contribution of a Dynamics, you first check if you already computed it by checking the map, otherwise you recompute it and add it to the Map for potential future calls.

Reply:
> as it happens during segment solving

Yeah for sure, we might be able to cache it now actually, will check

Reply:
+1 on calling it `get` to future proof. And contribution to keep it generic.
I would punt on the map, that is added complexity for some small (unquantified) performance improvement. If the user is doing multiple calls of `getContributions` let them do the caching.
+1 on having the user provide the dynamic.

We can thus have 2 methods:
- getContributions(frame) (just loops over all dynamics and calls the method below)
- getContribution(dynamic, frame) (which does it for specific dynamics, uses 95% of the code you have here)